### PR TITLE
feat: add new sample for analytics data (run_report with aggregations)

### PR DESCRIPTION
**Pull request by anweshan**
Adds sample for analytics data v1beta run_report with aggregations

### CODE REVIEW

**Review by ikuleshov — APPROVED** [review]

File: analyticsdata/src/run_report_with_aggregations.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+/**
+* Google Analytics Data API sample application demonstrating the usage of
+* metric aggregations in a report.
+* See https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/properties/runReport#body.request_body.FIELDS.metric_aggregations
+* for more information.
+
+* Usage:
+*   composer update
+*   php run_report_with_aggregations.php YOUR-GA4-PROPERTY-ID
+*/
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:27]:
```suggestion
 * Google Analytics Data API sample application demonstrating the usage of
 * metric aggregations in a report.
 * See https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/properties/runReport#body.request_body.FIELDS.metric_aggregations
 * for more information.
 * Usage:
 *   composer update
 *   php run_report_with_aggregations.php YOUR-GA4-PROPERTY-ID
 */
```

File: analyticsdata/src/run_report_with_aggregations.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+namespace Google\Cloud\Samples\Analytics\Data;
+
+// [START analyticsdata_run_report_with_aggregations]
+use Google\Analytics\Data\V1beta\BetaAnalyticsDataClient;
+use Google\Analytics\Data\V1beta\DateRange;
+use Google\Analytics\Data\V1beta\Dimension;
+use Google\Analytics\Data\V1beta\Metric;
+use Google\Analytics\Data\V1beta\MetricType;
+use Google\Analytics\Data\V1beta\MetricAggregation;
+
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:38]: needs PHPDoc
```suggestion
/**
 * @param string $propertyId
 */
```

File: analyticsdata/src/run_report_with_aggregations.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    MetricAggregation::MAXIMUM,
+    MetricAggregation::MINIMUM
+        ]
+    ]);
+
+    printRunReportResponseWithAggregations($response);
+}
+
+// Print results of a runReport call.
+function printRunReportResponseWithAggregations($response)
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:76]: Needs PHPDoc and typehint
```suggestion
/**
 * @param RunReportResponse $response
 */
function printRunReportResponseWithAggregations(RunReportResponse $response)
```

**Review by bshaffer — CHANGES_REQUESTED** [review]

File: analyticsdata/src/run_report_with_aggregations.php (hunk 3 above)
Comment (bshaffer) [reply analyticsdata/src/run_report_with_aggregations.php:76]: This also will require an import at the top for `RunReportResponse`

**Comment by anweshan** [comment]
@bshaffer the changes were made

File: analyticsdata/src/run_report_with_aggregations.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            new DateRange([
+                'start_date' => '365daysAgo',
+                'end_date' => 'today',
+            ]),
+        ],
+        'metricAggregations' => [
+    MetricAggregation::TOTAL,
+    MetricAggregation::MAXIMUM,
+    MetricAggregation::MINIMUM
+        ]
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:74]: fix whitespace
```suggestion
        'metricAggregations' => [
            MetricAggregation::TOTAL,
            MetricAggregation::MAXIMUM,
            MetricAggregation::MINIMUM
        ]
```

File: analyticsdata/src/run_report_with_aggregations.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+ */
+function printRunReportResponseWithAggregations($response)
+{
+    // [START analyticsdata_print_run_report_response_header]
+    printf('%s rows received%s', $response->getRowCount(), PHP_EOL);
+    foreach ($response->getDimensionHeaders() as $dimensionHeader) {
+        printf('Dimension header name: %s%s', $dimensionHeader->getName(), PHP_EOL);
+    }
+    foreach ($response->getMetricHeaders() as $metricHeader) {
+        printf('Metric header name: %s (%s)%s', $metricHeader->getName(), MetricType::name($metricHeader->getType()), PHP_EOL);
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:92]: more readable format:
```suggestion
        printf(
            'Metric header name: %s (%s)' . PHP_EOL,
            $metricHeader->getName(),
            MetricType::name($metricHeader->getType())
        );
```

File: analyticsdata/src/run_report_with_aggregations.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    }
+    // [END analyticsdata_print_run_report_response_header]
+
+    // [START analyticsdata_print_run_report_response_rows]
+    print 'Report result: ' . PHP_EOL;
+
+    foreach ($response->getRows() as $row) {
+        print $row->getDimensionValues()[0]->getValue()
+        . ' ' . $row->getMetricValues()[0]->getValue() . PHP_EOL;
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:101]: nit but more readable formatting
```suggestion
        printf(
            '%s %s' . PHP_EOL,
            $row->getDimensionValues()[0]->getValue()
            $row->getMetricValues()[0]->getValue()
        );
```

File: analyticsdata/src/run_report_with_aggregations.php [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    // [END analyticsdata_initialize]
+
+    // Make an API call.
+    $response = $client->runReport([
+        'property' => 'properties/' . $propertyId,
+        'dimensions' => [new Dimension(
+            [
+                'name' => 'country',
+            ]
+        )],
```
Comment (bshaffer) [inline analyticsdata/src/run_report_with_aggregations.php:58]: This is a little hard to follow. I might suggest breaking these into variables, such as
```php
$dimension = new Dimension(['name' => 'country']);
$metric = new Metric(['name' => 'sessions']);
$dateRange = (new DateRange())->setStartDate('365daysago')->setEndDate('today');
$metricAggregations = [
    MetricAggregation::TOTAL,
    MetricAggregation::MAXIMUM,
    MetricAggregation::MINIMUM
];
$response = $client->runReport([
    'property' => 'properties/' . $propertyId,
    'dimensions' => [$dimension],
    'metrics' => [$metric],
    'dateRanges' => [$dateRange],
]);
````

```

File: analyticsdata/test/analyticsDataTest.php [hunk 8]
```diff
@@ -50,4 +50,12 @@ public function testClientFromJsonCredentials()
             $this->assertStringContainsString('does-not-exist.json', $ex->getMessage());
         }
     }
+
+    public function testRunReportWithAggregations()
+    {
+        $propertyId = self::requireEnv('GA_TEST_PROPERTY_ID');
+        $output = $this->runFunctionSnippet('run_report_with_aggregations', [$propertyId]);
+
+        $this->assertRegExp('/Report result/', $output);
```
Comment (bshaffer) [inline analyticsdata/test/analyticsDataTest.php:59]: can we assert something more than just this? This is thin coverage for all the logic happening in the print test

Comment (anweshan) [reply analyticsdata/test/analyticsDataTest.php:59]: The purpose of the test is simply to test that the parameters used in the sample are indeed a valid query to analytics, rather than testing the logic of the analytics query itself. There isn't any static test data that exists that could test for example, that with the metric aggregations total, maximum, minimum that the report would look a certain way. But if all the parameters are valid syntax, the report will at least run and print out "Report result"

Also, the purpose of the analytics samples is to create samples representing different endpoints in consistency with the analytics samples for other languages such as Python, NodeJS, Java - where the tests also similarly test if the query runs and prints out Report Result, so we don't want inconsistency between the tests for samples

While I agree as a long term goal it would be great to create a static dataset that could be used for more comprehensive tests, as a minimum since it's just sample code, it's important to test that there aren't any typos/incorrect syntax, and that the report itself runs a valid query.

Comment (bshaffer) [reply analyticsdata/test/analyticsDataTest.php:59]: That seems fair to me, although I still think it'd be better to add a few more assertions, such as:

```php
$this->assertStringContainsString('rows received', $output);
$this->assertStringContainsString('Dimension header name:', $output);
$this->assertStringContainsString('Metric header name:', $output);
$this->assertStringContainsString('ReportResult:', $output);
```

**Review by bshaffer — APPROVED** [review]
